diff --git a/node_modules/next-auth/providers/dropbox.js b/node_modules/next-auth/providers/dropbox.js
index efabfee..39dfa9f 100644
--- a/node_modules/next-auth/providers/dropbox.js
+++ b/node_modules/next-auth/providers/dropbox.js
@@ -10,9 +10,22 @@ function Dropbox(options) {
     id: "dropbox",
     name: "Dropbox",
     type: "oauth",
-    authorization: "https://www.dropbox.com/oauth2/authorize?token_access_type=offline&scope=account_info.read",
+    authorization: "https://www.dropbox.com/oauth2/authorize?token_access_type=offline&scope=account_info.read files.metadata.read files.content.read",
     token: "https://api.dropboxapi.com/oauth2/token",
-    userinfo: "https://api.dropboxapi.com/2/users/get_current_account",
+    userinfo: {
+      request: async (context) => {
+        let res = await fetch(
+          "https://api.dropboxapi.com/2/users/get_current_account",
+          {
+            headers: {
+              Authorization: `Bearer ${context.tokens.access_token}`,
+            },
+            method: "POST",
+          }
+        );
+        return await res.json();
+      },
+    },
 
     profile(profile) {
       return {
